packages/NodeCollector/BinaryOpTreeRootLocator.php

```diff
@@ -20,20 +24,32 @@ final class BinaryOpTreeRootLocator
      */
     public function findOperationRoot(Expr $expr, string $binaryOpClass): Expr
     {
-        /** @var ?Expr $parentNode */
         $parentNode = $expr->getAttribute(AttributeKey::PARENT_NODE);
-        if ($parentNode === null || $parentNode::class !== $binaryOpClass) {
+        if (! $parentNode instanceof Node) {
             return $expr;
         }
 
-        assert($parentNode instanceof BinaryOp);
-        $isLeftChild = $parentNode->left === $expr;
-        $isRightChild = $parentNode->right === $expr;
-        $isRightChildAndNotBinaryOp = $isRightChild && $expr::class !== $binaryOpClass;
-        if ($isLeftChild || $isRightChildAndNotBinaryOp) {
+        if ($parentNode::class !== $binaryOpClass) {
+            return $expr;
+        }
+
+        if (! $parentNode instanceof BinaryOp) {
```

Comment:
This will never be `true` thanks to `$parentNode::class !== $binaryOpClass` check, the `assert` was there to make PHPStan realize this fact.